Core/Src/main.c

```diff
@@ -9,46 +9,57 @@
  * 
  */
 
-#include <string.h>
 #include <stdio.h>
-
-#include "stm32f0xx.h"
-#include "stm32f051_rcc.h"
-#include "stm32f051_gpio.h"
-#include "stm32f051_uart.h"
-#include "stm32f051_i2c.h"
+#include <stdlib.h>
+#include <string.h>
 
 #include "main.h"
 #include "pins.h"
 
 #include "ds3231.h"
+#include "uart_ring.h"
+
+struct
+{
+  char strOut[100];
+  uint8_t count;
+  char bufIn[100];
+} bufUART;
```

Comment:
I would suggest creating a separate program module to keep all UART related stuff: ring buffer, uart, and so on.

Reply:
I do not agree. One of the main tasks set for the software is the implementation of UART with a ring buffer. For this reason, I consider it advisable to leave it here. But maybe I'll clean it up when the OS appears